crates/nu-parser/src/flatten.rs

```diff
@@ -18,7 +19,7 @@ pub enum FlatShape {
     Custom(DeclId),
     DateTime,
     Directory,
-    External,
+    External(Span),
```

Comment:
Directly using the span here will grow the size of `FlatShape` by 8 bytes, not sure how much it matters, but could be an idea to box this (the `DeclID`/`VarID`s should be pointer sized)

Second thought, does it make sense to make this an `Option<Box<Span>>` just so it is more clear that we are dealing with the alias case or not? Then writing a unit test would be easier compared to having to sniff the unstable spans.